src/msg/simple/Pipe.cc

```diff
@@ -2461,7 +2461,7 @@ int Pipe::tcp_read(char *buf, int len)
     buf += got;
     //lgeneric_dout(cct, DBL) << "tcp_read got " << got << ", " << len << " left" << dendl;
   }
-  return len;
+  return 0;
```

Comment:
I am not a ceph specialist, but this seems curious. Why should this function return 0 in case of success? -1 is returned if ```got``` is lower zero. So wouldn't it be fine to return got and change return type of this function to ```ssize_t``` too?